components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/odata/ODataAdapter.java

```diff
@@ -886,96 +1023,174 @@ private Entity getNavigableEntity(ServiceMetadata metadata, EdmEntityType parent
 		for (DataColumn column : this.dataHandler.getTableMetadata().get(tableName).values()) {
 			CsdlProperty property = new CsdlProperty();
 			property.setName(column.getColumnName());
-			int columnType = column.getColumnType();
+			DataColumn.ODataDataType columnType = column.getColumnType();
 			switch (columnType) {
-				case Types.INTEGER:
+				case INT32:
 					property.setType(EdmPrimitiveTypeKind.Int32.getFullQualifiedName());
 					property.setNullable(column.isNullable());
+					property.setMaxLength(column.getMaxLength());
 					break;
-				case Types.TINYINT:
-					property.setType(EdmPrimitiveTypeKind.Int16.getFullQualifiedName());
-					property.setNullable(column.isNullable());
-				case Types.SMALLINT:
+				case INT16:
 					property.setType(EdmPrimitiveTypeKind.Int16.getFullQualifiedName());
 					property.setNullable(column.isNullable());
+					property.setMaxLength(column.getMaxLength());
 					break;
-				case Types.DOUBLE:
+				case DOUBLE:
 					property.setType(EdmPrimitiveTypeKind.Double.getFullQualifiedName());
 					property.setPrecision(column.getPrecision());
 					property.setScale(column.getScale());
 					property.setNullable(column.isNullable());
 					break;
-				case Types.VARCHAR:
-				case Types.CHAR:
-				case Types.LONGVARCHAR:
-				case Types.CLOB:
+				case STRING:
 					property.setType(EdmPrimitiveTypeKind.String.getFullQualifiedName());
 					property.setMaxLength(column.getMaxLength());
 					property.setNullable(column.isNullable());
-					property.setUnicode(false);
 					break;
-				case Types.BOOLEAN:
-				case Types.BIT:
+				case BOOLEAN:
 					property.setType(EdmPrimitiveTypeKind.Boolean.getFullQualifiedName());
 					property.setNullable(column.isNullable());
+					property.setMaxLength(column.getMaxLength());
 					break;
-				case Types.BLOB:
+				case BINARY:
 					property.setType(EdmPrimitiveTypeKind.Binary.getFullQualifiedName());
 					property.setNullable(column.isNullable());
+					property.setMaxLength(column.getMaxLength());
 					break;
-				case Types.BINARY:
-					property.setType(EdmPrimitiveTypeKind.Binary.getFullQualifiedName());
+				case BYTE:
+					property.setType(EdmPrimitiveTypeKind.Byte.getFullQualifiedName());
 					property.setNullable(column.isNullable());
+					property.setMaxLength(column.getMaxLength());
 					break;
-				case Types.LONGVARBINARY:
-				case Types.VARBINARY:
-					property.setType(EdmPrimitiveTypeKind.Binary.getFullQualifiedName());
+				case SBYTE:
+					property.setType(EdmPrimitiveTypeKind.SByte.getFullQualifiedName());
 					property.setNullable(column.isNullable());
+					property.setMaxLength(column.getMaxLength());
 					break;
-				case Types.DATE:
+				case DATE:
 					property.setType(EdmPrimitiveTypeKind.Date.getFullQualifiedName());
 					property.setNullable(column.isNullable());
+					property.setMaxLength(column.getMaxLength());
 					break;
-				case Types.DECIMAL:
-				case Types.NUMERIC:
+				case DURATION:
+					break;
```

Comment:
this was a mistake. I have fixed the duration case in https://github.com/wso2/carbon-data/commit/249c1f8e7a02ec26df519cad9408945b6322718b